### ISSUE TITLE
Use of undefined constant CAPTCHA_TYPE - assumed 'CAPTCHA_TYPE'

### ISSUE THREAD

**Issue by alivelimeli** [WoltLab/WCF#2469]
Hello,
I'm getting "Use of undefined constant CAPTCHA_TYPE - assumed 'CAPTCHA_TYPE'" error after fresh install.

```Error Type:
wcf\system\exception\ErrorException

Error Message:
Use of undefined constant CAPTCHA_TYPE - assumed 'CAPTCHA_TYPE'

File:
*/lib/system/WCF.class.php (313)

Stack Trace:
#0 */lib/system/request/Request.class.php (82):
wcf\system\WCF::handleError(8, 'Use of undefined constant CAPTCHA_TYPE - assumed \'CAPTCHA_TYPE\'', '*/lib/system/request/Request.class.php', 82, [ ])

#1 */lib/system/request/RequestHandler.class.php (94):
wcf\system\request\Request->execute()

#2 */acp/index.php (9):
wcf\system\request\RequestHandler->handle('wcf', true)
```

I've checked options.inc.php:

```php
<?php
/**
 * Default options.inc.php for package installation of package com.woltlab.wcf.
 * 
 * @author	Marcel Werk
 * @copyright	2001-2017 WoltLab GmbH
 * @license	GNU Lesser General Public License <http://opensource.org/licenses/lgpl-license.php>
 */
define('LAST_UPDATE_TIME', TIME_NOW);

define('COOKIE_PREFIX', 'wsc30_');
define('COOKIE_PATH', '');
define('COOKIE_DOMAIN', '');

define('HTTP_ENABLE_GZIP', 0);
define('HTTP_GZIP_LEVEL', 1);
define('HTTP_SEND_X_FRAME_OPTIONS', 0);

define('BLACKLIST_IP_ADDRESSES', '');
define('BLACKLIST_USER_AGENTS', '');
define('BLACKLIST_HOSTNAMES', '');

define('SESSION_TIMEOUT', 3600);
define('SESSION_VALIDATE_IP_ADDRESS', 0);
define('SESSION_VALIDATE_USER_AGENT', 0);

define('CACHE_SOURCE_TYPE', 'disk');
define('IMAGE_ADAPTER_TYPE', 'gd');
define('MODULE_MASTER_PASSWORD', 0);
define('TIMEZONE', 'Europe/Berlin');

define('ENABLE_DEBUG_MODE', 1);
define('ENABLE_BENCHMARK', 0);
define('EXTERNAL_LINK_TARGET_BLANK', 0);
define('URL_LEGACY_MODE', 0);
define('URL_TO_LOWERCASE', 1);
define('SEARCH_ENGINE', 'mysql');
define('SHOW_VERSION_NUMBER', 1);
define('LANGUAGE_USE_INFORMAL_VARIANT', 0);

define('WCF_OPTION_INC_PHP_SUCCESS', true);
```
And the options table is empty.

How can I solve it?

Thanks

**Comment by dtdesign** [comment]
Which version have you tried to install?

**Comment by alivelimeli** [comment]
I've tried 3.0.10 which is downloaded from: https://assets.woltlab.com/release/woltlab-suite-3.0.10.zip

**Comment by dtdesign** [comment]
This may be caused by overlapping cookies. This can happen if you have previously installed the software on the same domain, either by setting it to cover the entire domain (cookies set to `example.com`) and now try to install it on a subdomain (e. g. `www.example.com`). Or it may happen if you have cookies set via http, but are now trying to install the software via https/TLS, which confuses the browser's cookie manager.

Either way, the above error message is misleading, because it is the symptom of the aforementioned cookie mismatch. Try removing the cookies for the domain, remove the failed installation and try again.

Core 3.1 (currently in Beta, see https://community.woltlab.com/thread/263680-woltlab-suite-3-1-beta-3-download/) comes with an experimental change that utilizes ranomized cookie prefixes to avoid this issue entirely.

**Comment by alivelimeli** [comment]
I've removed all cookies & tried, result is same. I'm running it on 127.0.0.1 and it was a new installation from scratch to test your product.

I think the download link for woltlab suite on the community page is wrong. Its downloading forum 5.1.0 beta 3 zip file.

**Comment by dtdesign** [comment]
Actually the link is correct, we do not offer stand-alone builds of the Core during the Beta-/RC-phase, so it will always come bundled with one of the apps. You can uninstall the forums once the setup has completed though.

Do you experience the same issue with the Beta 3?

Also, you've said that you run it on `127.0.0.1`, are you using a non-default port?

**Comment by alivelimeli** [comment]
> Also, you've said that you run it on 127.0.0.1, are you using a non-default port?

Yes its running on 8080 port.

Also I've tried 3.1.0 or forum bundle from scratch couple of times and every time I get similar PHP constant error. 

Steps so far I've done;
- Downloaded 3.1.0 zip again
- Put into a new folder on 127.0.0.1:8080
- Created a new database
- Started installation with default steps (nothing changed except mysql connection details)
- And same error.

Also in one of the try (I think it was bundle) I got "undefined constant COOKIE_PREFIX - assumed 'COOKIE_PREFIX'" error.

**Comment by alivelimeli** [comment]
I've tried to install latest version (5.1.0 beta 3) and its installed without any problem. 

But 3.1.0 still not working (core wcf or forum fundle)

**Comment by alivelimeli** [comment]
For version 3.1.0, I've checked all PHP files, configuration files etc.

Those constants (CAPTCHA_TYPE or COOKIE_PREFIX) are not defined anywhere in the files. Probably its reading (or maybe generating) from options.xml file.

But as options.inc.php also doesn't includes those contants, WCF application throws Exception.

Is there a problem with options.inc.php ? Or maybe something else?

How can I solve for 3.1.0 version?

Thanks

**Comment by dtdesign** [comment]
Ignore the missing option or the `options.inc.php`, this has nothing to do with your problem. In fact, the cookie issues automatically trigger the login procedure, which is not functional at this point.

Long story short, the Core 3.0 version does not handle the port 8080 well at this point, so either stick with the Beta or use a default port (80 for HTTP, 442 for HTTPS/TLS).

**Comment by dtdesign** [comment]
Fixed in bbda3e4ffc9626b8e95d2dae1aa800c05ce4f26a
